connectivity/tests/errors.go

```diff
@@ -251,4 +251,10 @@ const (
 	failedToReleaseLock    = "Failed to release lock:"
 	previouslyUsedCIDR     = "Unable to find identity of previously used CIDR"                           // from https://github.com/cilium/cilium/issues/26881
 	klogLeaderElectionFail = "error retrieving resource lock kube-system/cilium-operator-resource-lock:" // from: https://github.com/cilium/cilium/issues/31050
+
+	// knownBugWireguardCollision is for a known issue: https://github.com/cilium/cilium/issues/31535.
+	// In spite of this occurrence, fqdn connectivity tests still pass thus it should be ok to ignore these for a while
+	// while we fix this issue.
+	// TODO: Remove this after: #31535 has been fixed.
+	knownBugWireguardCollision = "bind: address already in use" // from: https://github.com/cilium/cilium/issues/30901
```

Comment:
nit: Assuming this is not just a WireGuard problem, can we rename to `knownBugTransparentProxyPortCollision`?

I'd also prefer if we somehow checked for `Cannot forward proxied DNS lookup` too. `"address already in use" might be an error issues by other subsystems where it is probably more fatal

Reply:
Good point, I was working with the limitation that this has to be a single string.  We may want to support regex based matches for this.

Reply:
Kind of tangential, we should probably be able to override these exceptions with a flag rather than needing to open cli PRs...

Reply:
> nit: Assuming this is not just a WireGuard problem, can we rename to knownBugTransparentProxyPortCollision?

Left this as as wireguard as I reworked this to use a regex to capture a more specific exception.
If we start to see similar collision errors, but not related to the WG binding I think we should expose that for the time being.